src/System/Process/Typed.hs

```diff
@@ -624,7 +618,17 @@ startProcess pConfig'@ProcessConfig {..} = liftIO $ do
 
     pExitCode <- newEmptyTMVarIO
     waitingThread <- async $ do
-        ec <- P.waitForProcess pHandle
+        ec <-
+          if multiThreadedRuntime
+            then P.waitForProcess pHandle
+            else
+              let loop = do
+                    threadDelay 20000
```

Comment:
For our records, we also discussed whether we should use the pipe approach to notice when the child process has _not_ yet shut down in the case that the user doesn't give the flag that emulates `CLOEXEC` behaviour for FDs before forking (e.g. pipe closes => process may be dead or just have closed that FD, pipe open => process must be alive), but chose to not add that extra complexity / possibly difficult code path to keep things easier to debug.